### PR TITLE
feat: add transaction support

### PR DESCRIPTION
**Pull request by manakuro**
Add transaction support

Related: #4

### CODE REVIEW

File: usecase/interactor/user_interfactor.go [hunk 1]
```diff
@@ -30,7 +32,23 @@ func (us *userInteractor) Get(u []*model.User) ([]*model.User, error) {
 }
 
 func (us *userInteractor) Create(u *model.User) (*model.User, error) {
-	u, err := us.UserRepository.Create(u)
+	data, err := us.DBRepository.Transaction(func(i interface{}) (interface{}, error) {
+		u, err := us.UserRepository.Create(u)
```
Comment (zbysir) [inline usecase/interactor/user_interfactor.go:36]: It do not use `i` to exec sql, are you sure it will work?

Comment (manakuro) [reply usecase/interactor/user_interfactor.go:36]: Yes, the transaction commits here.

https://github.com/manakuro/golang-clean-architecture/pull/6/files#diff-a7cdc9f4ff852ba7dce8cb18e11879dd3c25a69b1f102d6b2c59a5193fd2b2a9R35

Comment (zbysir) [reply usecase/interactor/user_interfactor.go:36]: Thank you for your (fast) answer , so what is `i` used for?

Comment (manakuro) [reply usecase/interactor/user_interfactor.go:36]: The `i` is the database object generated here.

https://github.com/manakuro/golang-clean-architecture/pull/6/files#diff-a7cdc9f4ff852ba7dce8cb18e11879dd3c25a69b1f102d6b2c59a5193fd2b2a9R20

So if you want to `commit` or `rollback` manually, you can use this like so:

```.go
i.Rollback()
```

But for now `i` is just an interface type so it needs to be typed correctly. 

Thanks for your question!

Comment (zbysir) [reply usecase/interactor/user_interfactor.go:36]: Sorry for not being very accurate, I mean, I think it is problematic to execute SQL without using `i`, maybe you can write a small test case to test it.

I ran the following code and found that it did not rollback
```
us.UserRepository.Create(u)

i.Rollback()
```

Comment (manakuro) [reply usecase/interactor/user_interfactor.go:36]: Oh, okay. I'll look into it. Thanks for reporting!

Comment (juddbaguio) [reply usecase/interactor/user_interfactor.go:36]: any updates on this one? @manakuro

**Pull request merged**